Organization repos

nickdiorio opened the pull request:
The existing implementation implicitly assumes that the username is the same as the organization which owns the repository, but this isn't always the case.  As an example, corporate, government, or other organizations often have their own GitHub organization which result in repo names like:

`https://github.com/NREL/SAM`

Where NREL is an organization with many users.  For these repo's, supplying a username/password is not enough, since the existing implementation would create:

`https://github.com/nickdiorio/SAM`

For my username as an example.  This pull request adds the flexibility to specify an organization in addition to the username/password, i.e:

`python main.py 'NREL' 'nickdiorio' 'SAM' 'save_csv'`

When the organization and username are the same, my implementation would require:

`python main.py 'nickdiorio' 'nickdiorio' 'github-traffic-stats' 'save_csv' `

Which could perhaps be made more elegant by making the `organization` input default to `username`.

Hope this helps, happy to revamp if desired.

nchah commented:
Hi Nick,

Thank you for your contribution! This is certainly useful for the use case you've mentioned.

I've tested it out and had no issue with typing out the organization and username arguments in each call. For the cases where there is no "organization", what are your thoughts on moving it to the end as an optional argument? If this is too onerous the current implementation looks great too.

@AnthonyBloomer - Anthony has helped put this on PyPI so it would be great to hear his thoughts. Thank you!

nickdiorio commented:
@nchah , sure, no problem, thanks for creating a very useful tool!

I think it makes sense to make "organization" an optional argument.  I'll do that and add to this PR.

Best,
Nick

AnthonyBloomer commented:
Hi guys,

@nickdiorio, this is a nice addition. 

If possible could you update the README with the additional usage information?

nickdiorio commented:
@AnthonyBloomer , @nchah , I've updated to specify organization as an optional parameter and added to the README.  I also included an example that I've been using for reference.  I can also make it more generic, or if you'd like to modify, no worries.  Just let me know if you'd like any other changes.

nchah commented:
Thank you @nickdiorio !! Merging it in now. :)

nchah commented:
@AnthonyBloomer - please mention if you need anything more for updates to the PyPI page https://pypi.python.org/pypi/github-traffic-stats. I've also sent you an invite to be a collaborator on this repo if that helps.

It's a pleasure to work with you guys through GitHub. Thanks!
